Parser/string_parser.c

```diff
@@ -21,9 +21,11 @@ warn_invalid_escape_sequence(Parser *p, const char *first_invalid_escape, Token
     if (msg == NULL) {
         return -1;
     }
-    if (PyErr_WarnExplicitObject(PyExc_DeprecationWarning, msg, p->tok->filename,
+    PyObject *category = (octal ? PyExc_DeprecationWarning
+                                : PyExc_SyntaxWarning);
```

Comment:
Not sure whether this is applicable here and it is surely a nit, but should this be taking `feature_version` into consideration. What should for example `ast.parse("f'\A'", feature_version=(3, 10))` warn with, a deprecation or a syntax warning?

In general, it kinda feels that `feature_version` adds more maintenance cost than benefit.

Reply:
I added a test on feature_version.